enforcer_cached_test.go

```diff
@@ -32,21 +32,27 @@ func TestCache(t *testing.T) {
 	testEnforceCache(t, e, "alice", "data2", "read", false)
 	testEnforceCache(t, e, "alice", "data2", "write", false)
 
-	// The cache is enabled, so even if we remove a policy rule, the decision
-	// for ("alice", "data1", "read") will still be true, as it uses the cached result.
```

Comment:
Should add some comment to explain our new mechanism and why it works for this test case.